internal/tlssession/tlssession.go

```diff
@@ -144,7 +141,6 @@ func (ws *workersState) doTLSAuth(conn net.Conn, config *tls.Config, errorch cha
 		errorch <- err
 		return
 	}
-	//defer tlsConn.Close() // <- we don't care since the underlying conn is a tlsBio
```

Comment:
I'd keep this comment because otherwise someone reading may wonder why we don't close the Conn.